apps/api/src/domain/models/types/HasEntityId.ts

```diff
@@ -0,0 +1,9 @@
+import { EntityId } from '../../types/ResourceId';
+
+export type HasEntityID<T = undefined> = T extends undefined
```

Comment:
Would it be better to use an interface for this? Our Resource and Context models could then implement the interface.

Reply:
done